### DISCUSSION TITLE
TypeScript migration

### DISCUSSION THREAD

**Discussion by pawel975** [gbowne1/reactsocialnetwork#157] [Ideas]
I wonder if migration to TypeScript would make app easier to maintain in the long run. We have propTypes so far but all TypeScript features could push it further. What do you think guys? @gbowne1 @manuel12. In my opinion migration would be easier right know since app is not so big at the moment.

**Comment by gbowne1** [comment]
It might be easier right now.  However, My only concerm is me not being able to contribute as I have not taken the time to learn contribute.  And, I would wonder if it would help or hinder new contributions.

**Reply by pawel975** [reply]
Hmm, you're right it could scared potential new contributors.

**Reply by gbowne1** [reply]
I mean its an excellent idea @pawel975 .  Im just comcerned for those who dont know or use Typescript or understand types or type safety.

**Reply by manuel12** [reply]
I have the same feeling. I would like the opportunity to finally try and learn Typescript, but it might be a barrier to entry.

**Reply by gbowne1** [reply]
I want to learn too.  I also feel like we don't have enough working panels && components to even get started on Typescript.  I will mark it as a future todo.  Because I do feel like it would be good once we have a useable product.

**Reply by pawel975** [reply]
We could also implement TypeScript partially, TS and JS can work together and only new components would be written in TS. It would provide type checking for this component but won't be problem for newcomers because they don't have to use TS for new functionalities.

**Reply by gbowne1** [reply]
@pawel975 I would only do that where it makes the most sense initially.

**Comment by gbowne1** [comment]
@dalekvim
